### PR TITLE
Feature/31 finish sign up form

### PR DESCRIPTION
**Pull request by deny1994**

### CODE REVIEW

File: backend/src/index.js [hunk 1]
```diff
@@ -49,6 +49,7 @@ const typeDefs = gql`
     signup(
       email: String!
       password: String!
+      passwordConfirmation: String
```
Comment (xkadl17) [inline backend/src/index.js:52]: I think we do not need to send the confirmation

File: frontend/src/pages/SignUpPage.js [hunk 2]
```diff
@@ -1,29 +1,45 @@
...
+    $name: String!
     $password: String!
     $passwordConfirmation: String!
+    $username: String!
   ) {
     signup(
       email: $email
+      name: $name
       password: $password
       passwordConfirmation: $passwordConfirmation
```
Comment (xkadl17) [inline frontend/src/pages/SignUpPage.js:20]: we do not need to send password confirmation

File: frontend/src/pages/SignUpPage.js [hunk 3]
```diff
@@ -1,29 +1,45 @@
...
+  const auth = useAuth();
+  const history = useHistory();
   const [signupRequest, signupRequestState] = useMutation(SIGNUP_MUTATION, {
-    onCompleted: ({ signup: { email } }) => {
-      console.log(email);
+    onCompleted: ({ signup: { user, token } }) => {
+      auth.signin({ token, user });
+      history.replace('/');
     },
+    onError: () => {},
```
Comment (xkadl17) [inline frontend/src/pages/SignUpPage.js:42]: remove unused cb

**Review by xkadl17 — APPROVED** [review]

**Review by okrsnak — APPROVED** [review]

File: frontend/src/atoms/UserUsername.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
+export function UserUsername({ username }) {
```
Comment (heracek) [inline frontend/src/atoms/UserUsername.js:3]: I'd use `UserUserName`.

There is already existing `UserName` component, so let's keep it consistent.

Comment (heracek) [reply frontend/src/atoms/UserUsername.js:3]: lowercase `username` is ok

**Review by heracek — APPROVED** [review]
Looks great, good job 👍 

See comments for minor issues.

**Comment by heracek** [comment]
Budu to potrebovat mergnout behem dnesniho vecera.
Tak ty komentare opravim, pushnu to a mergnu.

File: frontend/src/pages/SignUpPage.js (hunk 3 above)
Comment (heracek) [reply frontend/src/pages/SignUpPage.js:40]: This is actually required. Without it you receive "unhandled exception" errors.

**Pull request merged**